core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala

```diff
@@ -223,16 +235,45 @@ abstract class EndToEndAuthorizationTest extends IntegrationTestHarness with Sas
   private def setWildcardResourceAcls() {
     AclCommand.main(produceConsumeWildcardAclArgs)
     servers.foreach { s =>
-      TestUtils.waitAndVerifyAcls(TopicReadAcl ++ TopicWriteAcl ++ TopicDescribeAcl ++ TopicBrokerReadAcl, s.apis.authorizer.get, wildcardTopicResource)
+      TestUtils.waitAndVerifyAcls(TopicReadAcl ++ TopicWriteAcl ++ TopicDescribeAcl ++ TopicCreateAcl ++ TopicBrokerReadAcl, s.apis.authorizer.get, wildcardTopicResource)
       TestUtils.waitAndVerifyAcls(GroupReadAcl, s.apis.authorizer.get, wildcardGroupResource)
     }
   }
 
-  protected def setAclsAndProduce() {
-    AclCommand.main(produceAclArgs)
-    AclCommand.main(consumeAclArgs)
+  def testProduceConsumeTopicAutoCreateTopicCreateAcl(): Unit = {
+    // topic2 is not created on setup()
+    val tp2 = new TopicPartition("topic2", 0)
+    setAclsAndProduce(tp2)
+    consumers.head.assign(List(tp2).asJava)
+    consumeRecords(this.consumers.head, numRecords, topic = tp2.topic)
+  }
+
+  @Test
+  def testProduceConsumeTopicAutoCreateClusterCreateAcl(): Unit = {
```

Comment:
Do we need new tests here given `AuthorizerIntegrationTest`.

Reply:
yes, we think so because EndToEndAuthorizationTest uses the acl tool command line

Reply:
Don't we have `AclCommandTest` for that?

Reply:
Generally, this test was meant to cover some high level end to end cases. It seems OK to add an auto create case, but overkill to add one for Create Topic and another for Create Cluster. We should add a test to `AclCommandTest` for the latter.

Reply:
Thanks, I see the `AclCommandTest` has already coverage of adding cluster create and topic create acls. The `AuthorizerIntegrationTest` covers all request handling logic with and without auto-create.

So we only left one end2end test case for auto create with topic create acl